decorator.rb

```diff
@@ -0,0 +1,13 @@
+require './nameable'
+
+# Create base Decorator
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
```

Comment:
- [ ] You did a great job, but for the correction, I would like to tell you that you don't have any constructor for the `Nameable` so that's why you don't need to pass `super()` because your `Nameable` functionalities are already inherited by the `Decorator` class, that's why please remove it and please make a `main.rb` file and put the below content for ease of access so that anyone can see and run your project without any problem.

```ruby
person = Person.new(22, 'maximilianus')
 person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
 capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
 capitalizedTrimmedPerson.correct_name
```

Reply:
thank you i will refactor and create main.rb and app.rb to make everything clear for the user . 👍